src/lib/common/sol-platform-impl-linux-micro.c

```diff
@@ -1146,9 +1146,30 @@ sol_platform_impl_set_timezone(const char *timezone)
     if (symlink(path, "/etc/localtime") < 0) {
         SOL_WRN("Could not create the symlink to the timezone %s", timezone);
         return -errno;
+    } else {
+        /* Check if it was linked to the right place to prevent TOCTOU */
+        char buf[PATH_MAX];
+        ssize_t len;
+
+        if ((len = readlink("/etc/localtime", buf, sizeof(buf)-1)) < 0) {
```

Comment:
Nevermind, didn't see the `-1` there. Missing the spaces to ensure uncrustify is happy.